src/main/java/de/zalando/ep/zalenium/proxy/AutoStartProxySet.java

```diff
@@ -600,3 +645,34 @@ public String toString() {
     }
 
 }
+
+class TrackedConcurrentHashMap<K, V> extends ConcurrentHashMap<K, V> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TrackedConcurrentHashMap.class.getName());
+
+    @Override
+    public V remove(Object key) {
+        StringBuilder stackTrace = new StringBuilder();
+        Arrays.asList(Thread.currentThread().getStackTrace())
+            .forEach( (val) -> stackTrace.append(val));
+
+        LOGGER.info(String.format("TRACK: Remove proxy %s\n%s",
+                key.toString(),
+                stackTrace.toString()));

```

Comment:
This looks like debugging to me, I think it should use LOGGER.debug, and then wrap in an if statement checking if debug is enabled, because getting stack traces is usually pretty expensive.

Reply:
Yes, if was for debugging, we're going to remove it